File: leetcode/2490-circular-sentence/solution.cpp
/**
 * Author: 1chooo<hugo970217@gmail.com>
 * Problem: https://leetcode.com/problems/circular-sentence
 * Runtime: 0ms (100.00%)
 */

const static auto _ = []() {
    cin.tie(nullptr)->sync_with_stdio(false);
    return nullptr;
}();

class Solution {
  public:
    bool isCircularSentence(string sentence) {
        int len = sentence.size();

        if (sentence[0] != sentence[len - 1])
            return false;

        for (int i = 1; i < len - 1; i++) {
            if (sentence[i] == ' ' &&
                sentence[i - 1] != sentence[i + 1])
                return false;
        }

        return true;
    }
};

File: leetcode/2490-circular-sentence/solution.py
"""
* Author: 1chooo<hugo970217@gmail.com>
* Problem: https://leetcode.com/problems/circular-sentence
* Runtime: 0ms (100.00%)
"""

class Solution:
    def isCircularSentence(self, sentence: str) -> bool:
        sentence_len: int = len(sentence)

        if (sentence[0] != sentence[-1]):
            return False
        
        for i in range(1, sentence_len):
            if (sentence[i] == ' ' and 
                sentence[i - 1] != sentence[i + 1]):
                return False
            
        return True
